### PR TITLE
Driver guide/examples: fix an error and improve discoverability

### PR DESCRIPTION
**Pull request by mpg**
## Description

Make the document that describes the current reality of driver development more discoverable, and fix an error in it.

(See also [this thread on slack](https://arm-ce.slack.com/archives/C01CC4Y1T1P/p1691400500094489).)

## PR checklist

- [x] **changelog** not required - documentation only
- [x] **backport** not required - that document is not in 2.28
- [x] **tests** not required - documentation only

### CODE REVIEW

File: docs/proposed/psa-driver-wrappers-codegen-migration-guide.md [hunk 1]
```diff
@@ -2,6 +2,7 @@ Migrating to an auto generated psa_crypto_driver_wrappers.c file
 ================================================================
 
 **This is a specification of work in progress. The implementation is not yet merged into Mbed TLS.**
+For a description of the current state of drivers Mbed TLS, see our [PSA Cryptoprocessor driver development examples](../psa-driver-example-and-guide.html).
```
Comment (gilles-peskine-arm) [inline docs/proposed/psa-driver-wrappers-codegen-migration-guide.md:5]: This document describes the generator script as it is now, so the line about a work in progress is not correct. When the generator script is completed, this document should no longer be needed: the integration guide should suffice.

File: docs/proposed/psa-driver-integration-guide.md [hunk 2]
```diff
@@ -2,6 +2,7 @@ Building Mbed TLS with PSA cryptoprocessor drivers
 ==================================================
 
 **This is a specification of work in progress. The implementation is not yet merged into Mbed TLS.**
+For a description of the current state of drivers Mbed TLS, see our [PSA Cryptoprocessor driver development examples](../psa-driver-example-and-guide.html).
```
Comment (gilles-peskine-arm) [inline docs/proposed/psa-driver-integration-guide.md:5]: We should add this line to `psa-driver-interface.md` as well.

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
I think one of the lines was added to the wrong file

**Review by gilles-peskine-arm — APPROVED** [review]
LGTM

**Comment by davidhorstmann-arm** [comment]
Did the links to the driver example guide ever work?

We've just had them reported as broken in #8453 and I think perhaps they should point to `.md` not `.html`.

**Comment by gilles-peskine-arm** [comment]
The links to `.html` files work if you build the HTML collection locally. They don't work if you browse the `.md` files on GitHub.

**Pull request merged**